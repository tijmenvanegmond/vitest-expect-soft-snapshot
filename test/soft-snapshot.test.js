import { it, expect } from 'vitest'

it('expect.soft doesnt stops execution on failure', () => {
    console.log('Asserting object with expect.soft, should fail on timestamp...')
    expect.soft({foo:"bar"}).toMatchObject({foo:"baz"})
    // continues here
    console.log(`Should log and does`)
    expect.soft(true).toBe(false)
})

it('expect.soft with toMatchSnapshot stops execution on failure', () => {    
    // These should fail but with soft assertions, execution should continue
    // However, toMatchSnapshot does not work properly with expect.soft
    expect.soft({time: new Date()}).toMatchSnapshot()
    // Doesnt continue here
    console.log(`Should log but doesn't`)
    // This assertion should still run even if the snapshots above fail but doesn't
    expect.soft(true).toBe(false)
})