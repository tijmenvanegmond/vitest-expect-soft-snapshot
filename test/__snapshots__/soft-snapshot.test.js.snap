// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`expect.soft with toMatchSnapshot stops execution on failure > content-v1 1`] = `
{
  "data": "This is some content",
  "timestamp": "2025-10-07T09:16:19.273Z",
}
`;
